llvm/utils/TableGen/VarLenCodeEmitterGen.cpp

```diff
@@ -505,7 +506,16 @@ std::string VarLenCodeEmitterGen::getInstructionCaseForEncoding(
   if (!PostEmitter.empty())
     SS.indent(I) << "Inst = " << PostEmitter << "(MI, Inst, STI);\n";
 
-  return Case;
+  // Resize the scratch buffer if it's to small.
+  std::string ScratchResizeStr = "";
+  if (VLI.size() && !VLI.isFixedValueOnly()) {
+    raw_string_ostream RS(ScratchResizeStr);
+    std::string High = utostr(HighScratchAccess);
```

Comment:
I think `raw_string_ostream` can convert unsigned values to strings. So maybe
```
<< "if (Scratch.getBitWidth() < " << High << ") "
```
would be better than calling `utostr` here.